nada_dsl/nada_types/scalar_types.py

```diff
@@ -2,7 +2,7 @@
 """The Nada Scalar type definitions."""
 
 from dataclasses import dataclass
-from typing import Union
+from typing import Any, Self, Union, TypeVar
```

Comment:
Self is a feature added in 3.11, we cannot use this feature until we move up the minimum version.

Reply:
I think it should be enough to import it from `typing_extensions`, but I haven't tested it yet -- my python dev environment is extremely finicky.

Reply:
well, you should add it to the list of dependencies in pyproject.toml then. I didn't see the change and therefore I assumed you weren't using anything new.

Reply:
Hopefully that would fix the build